library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java

```diff
@@ -373,20 +376,41 @@ public void seekTo(final int msec) {
             case STATE_PREPARED:
             case STATE_PAUSED:
             case STATE_PLAYBACK_COMPLETED:
-                mLock.lock();
                 if (mTrack == null) {
                     return;
                 }
                 mTrack.flush();
-                mExtractor.seekTo(((long) msec * 1000), MediaExtractor.SEEK_TO_PREVIOUS_SYNC);
-                Log.d(TAG, "seek completed, position: " + (mExtractor.getSampleTime() / 1000L));
-                if (owningMediaPlayer.onSeekCompleteListener != null) {
-                    owningMediaPlayer.onSeekCompleteListener.onSeekComplete(owningMediaPlayer);
-                }
-                mLock.unlock();
-                if (playing) {
-                    start();
-                }
+
+                final boolean wasPlaying = playing;
+
+                Thread t = new Thread(new Runnable() {
```

Comment:
Why is the player now seeking in another thread?
Is ``MediaExtractor`` even thread-safe?